File: src/main/java/meetme/tweets/TweetResource.java
package meetme.tweets;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.Collection;
import java.util.List;

@RestController
public class TweetResource {

    @Autowired
    TweetsService tweetsService;

    private void createNewTweet() {
    }

    @PostMapping("tweets/follow")
    @CrossOrigin
    private void follow(@RequestBody Following following) {
        tweetsService.follow(following);
    }

    @CrossOrigin
    @GetMapping("followers/{userId}")
    private List<Followers> showFollowers(@PathVariable String userId) {
       return tweetsService.findFollowers(userId);
    }


    @CrossOrigin
    @GetMapping("followers")
    private Collection<Followers> getAllFollowers() {
        return (Collection<Followers>) tweetsService.findAll();
    }


    @PostMapping("tweets/tweet")
    @CrossOrigin
    private void tweet(@RequestBody Tweet tweet) {
        tweetsService.tweet(tweet);
    }


    private void unFollow() {
    }

    private void showTimeline() {
    }


}

File: src/main/java/meetme/tweets/TweetResource.kt
package meetme.tweets

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
class TweetResource {

    @Autowired
    lateinit var tweetsService: TweetsService;


    private val allFollowers: Collection<Followers>
        @CrossOrigin
        @GetMapping("followers")
        get() = tweetsService!!.findAll() as Collection<Followers>

    @PostMapping("tweets/follow")
    @CrossOrigin
    private fun follow(@RequestBody following: Following) = tweetsService!!.follow(following)

    @CrossOrigin
    @GetMapping("followers/{userId}")
    private fun showFollowers(@PathVariable userId: String) = tweetsService!!.findFollowers(userId)

    @PostMapping("tweets/tweet")
    @CrossOrigin
    private fun tweet(@RequestBody tweet: Tweet): Tweet = tweetsService!!.tweet(tweet)


    private fun unFollow() {}

    @GetMapping("tweets/all")
    @CrossOrigin
    private fun getAllTweets() = tweetsService!!.findAll()
}
